addons/crm/models/crm_team.py

```diff
@@ -463,7 +473,13 @@ def _allocate_leads(self, creation_delta_days=7):
 
             # assign + deduplicate and concatenate results in teams_data to keep some history
             candidate_lead = teams_data[team]["leads"][0]
-            assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            try:
+                assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            except Exception as e:
+                if auto_commit:
+                    _logger.error("[Assign Lead] Error while assigning lead %s (ID: %d) to team %s (ID: %d). Error: %s", candidate_lead.name, candidate_lead.id, team.name, team.id, str(e))
```

Comment:
Last one then we're good, I'll just try it locally for sanity then green it

Would always log "which lead caused the error", not just if auto-commit :)
Same for members try/catch